File: src/Java/733.FloodFill.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;

import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int newColor) {
        LinkedList<HashMap.Node<Integer, Integer>> site = new LinkedList<>();
        int val = image[sr][sc];
        image[sr][sc] = newColor;
        site.push(pair<int,int>(sr, sc));
        while(!site.empty()){
            int x = site.front().first, y = site.front().second;
            site.pop();
            if(x > 0 && image[x-1][y] == val && image[x-1][y] != newColor){
                site.push(pair<int, int>(x - 1, y));
                image[x-1][y] = newColor;
            }
            if(x < image.size() - 1 && image[x+1][y] == val && image[x+1][y] != newColor){
                site.push(pair<int, int>(x + 1, y));
                image[x+1][y] = newColor;
            }
            if(y > 0 && image[x][y-1] == val && image[x][y-1] != newColor){
                site.push(pair<int, int>(x, y - 1));
                image[x][y-1] = newColor;
            }
            if(y < image[0].size() - 1 && image[x][y+1] == val && image[x][y+1] != newColor){
                site.push(pair<int, int>(x, y + 1));
                image[x][y+1] = newColor;
            }
        }
        return image;
    }
}

File: src/CSharp/701-800/733.FloodFill.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace PeefyLeetCode.FloodFill
{
    public class Solution
    {
        public int[,] FloodFill(int[,] image, int sr, int sc, int newColor)
        {
            int oldColor = image[sr, sc];
            if (newColor == oldColor)
                return image;
            image[sr, sc] = newColor;
            if (sr - 1 >= 0 && image[sr - 1, sc] == oldColor)
            {
                image = FloodFill(image, sr - 1, sc, newColor);
            }
            if (sr + 1 < image.GetLength(0) && image[sr + 1, sc] == oldColor)
            {
                image = FloodFill(image, sr + 1, sc, newColor);
            }
            if (sc - 1 >= 0 && image[sr, sc - 1] == oldColor)
            {
                image = FloodFill(image, sr, sc - 1, newColor);
            }
            if (sc + 1 < image.GetLength(1) && image[sr, sc + 1] == oldColor)
            {
                image = FloodFill(image, sr, sc + 1, newColor);
            }
            return image;
        }
    }
}

